enhancements/network/host-level-openvswitch.md

```diff
@@ -0,0 +1,138 @@
+
+## Motivation
+
+Firstly, running OVS as a SystemD unit matches the technical reality:
+1. Only one ovs-vswitchd process can run on a node. Multitenancy is built in to the daemon, not the kernel.
+2. ovs-vswitchd is tightly coupled, protocol-wise, with the kernel
+3. ovs-vswitchd is in the critical path for networking. Stopping it blocks **all new flows**.
+
+Thus, the proposed solution has the following advantages:
+1. Simplifies the burden on OVS maintainers by shrinking the test matrix of kernel-daemon versions.
+2. Removes a known source of packet loss during upgrades
+3. Allows OVS to be shared between multiple networking components (openshift-sdn, CNV).
+
+### Goals & Requirements
+
+1. OVS is run in a SystemD unit provided by the base OS, but is still manageable as if it were a Kubernetes Pod:
+  - kubectl logs works
+  - Liveness- and Readiness- probes reflect the unit’s state
+  - Note: it is a non-goal that deleting the sidecar Pod stops OVS
+2. OVS is stable and more available than the current state.
+
+### Non-Goals
+
+1. Provide a generic solution for managing SystemD units via Kubernetes
+2. Reconcile all possible configuration changes (e.g. stopping / disabling OVS).
+
+## Proposal
+
+There is a pod that acts as a “sidecar” or “babysitter” for the openvswitch systemd unit. It needs to
+
+- Ensure the OVS unit is enabled and started
+- Provide a Liveness probe
+- Provide a Readiness probe
+- Apply any desired OVS configuration (loglevel, memory usage, etc)
+- Print log messages to stdout
+
+Configuration for OVS is always presented as a set of key-value pairs that are inserted in to a configuration database. If needed, we should allow custom configuration to be expressed as a ConfigMap, which is reconciled to the database.
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+As a prototype, this can be implemented with just a few shell scripts. Ideally, it should be ultimately implemented as a go program that talks to the SystemD DBus API directly.
+
+
+### Risks and Mitigations
+
+The biggest risk is that OVS is not running when we need it. The nice dependency management and ordering provided by SystemD doesn’t help us here. It could happen that, for whatever reason, the OVS consumer is scheduled and the OVS sidecar is not. One simple mitigation is PodAffinity. Or, we could include the babysitter process as a container inside ovn-kubernetes node daemonset, rather than a separate daemonset.
+
+The second risk is one of adopting existing clusters. How we migrate from containerized to non-containerized openvswitch will require careful planning.
+
+Not to be discounted is administrative churn. We disrupted many administrators’ workflows by moving OVS from a systemd service to a container. Now we’re reversing that decision - albeit for good reason. The mitigation is to try and merge expectations.
```

Comment:
I think providing the reference to the propsoal (running OVS as deamonset) would be helpful for the reviewers.